pkg/storage/sst.go

```diff
@@ -263,11 +272,10 @@ func UpdateSSTTimestamps(
 		} else if !ok {
 			break
 		}
-		if iter.UnsafeKey().Timestamp.IsEmpty() {
-			return nil, errors.New("inline values or intents are not supported")
-		}
-		if len(iter.UnsafeValue()) == 0 {
-			return nil, errors.New("SST values cannot be tombstones")
+		key := iter.UnsafeKey()
+		if key.Timestamp != from {
+			return nil, errors.Errorf("unexpected timestamp %s (expected %s) for key %s",
+				key.Timestamp, from, key.Key)
 		}
```

Comment:
For my knowledge, why did we remove tombstone check, is it because we already verify that in assertSSTContents and there is nothing wrong at rewriting tombstone timestamp from "sst validity" perspective?

Reply:
This check already covers it. A tombstone has an empty timestamp, and we reject empty `from` timestamps at the start of this function, so a tombstone's timestamp would not equal `from` and therefore trigger this error. And yes, we also check it in `assertSSTContents`.

Reply:
Sorry, I was thinking about inline values, which we also used to check here.

You're right, we don't check tombstones here (empty value), because that doesn't have anything to do with `UpdateSSTTimestamps()` -- the job of this function is just to rewrite timestamps, and it doesn't care about the values. We do general `AddSSTable` validation in `assertSSTContents`.